src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs

```diff
@@ -295,13 +295,25 @@ private static void PopulateAllSystemTimeZones(CachedData cachedData)
 
             if (Invariant)
             {
-                return;
+                return cachedData._systemTimeZones;
             }
 
+            // The filtered list that shouldn't have any duplicates.
+            Dictionary<string, TimeZoneInfo> filteredTimeZones = new Dictionary<string, TimeZoneInfo>(StringComparer.OrdinalIgnoreCase)
```

Comment:
Presumably we expect it to have very few duplicates? I'm wondering if we should pass cachedData._systemTimeZones.Count as the capacity to the ctor.

Reply:
This code execute when populating all TZ. I expect in main stream cases `cachedData._systemTimeZones.Count` will have very small number at that time as a few time zones will be used at that time. But it is a good idea to use some reasonable initial capacity for main stream cases. I'll check that and wil come with some reasonable number.